config/label-management.md

```diff
@@ -0,0 +1,126 @@
+python3.8 -m venv venv
+source venv/bin/activate
+python -m pip install --upgrade pip wheel
+python -m pip install -r requirements.txt
+```
+
+1. Create a local configuration file by running `touch config`
+1. Open `config` in your favorite editor for editing
+1. Create a GitHub [Personal access token](https://github.com/settings/tokens)
+
+- Give it a descriptive name like "label management"
+- **Only** select the `repo` scope
+- Copy the generated token into `GITHUB_USER_TOKEN`
+
+```sh
+export GITHUB_USER_TOKEN="<YOUR_TOKEN_HERE>"
+export LABELS_TOKEN=${GITHUB_USER_TOKEN}
+export LABELS_USERNAME="<YOUR_USERNAME>"
+export LABELS_ORGNAME="cisagov"
+```
+
+1. Save your changes and run `source config`
+
+## Label management operations ##
+
+There are several operations you'll probably want to do, now that you've
+set up the environment and have the script ready.
+
+### Re-initializing the environment ###
+
+When you return to the directory in a new shell, remember to:
+
+```console
+source venv/bin/activate
+source config
+```
+
+### List all organization repositories ###
+
+```console
+python list-all-repos.py
+```
+
+This will yield a list of all repositories you have access to in the specified
+GitHub organization. If you have not specified a token, this will yield a list
+of all public repositories.
+
+### Back up all current repository labels ###
+
+This will generate a `.toml` file for each repository in a directory of your
+choosing. Each `.toml` file will contain the details of the repository's
+labels, including names, colors, and descriptions.
+
+Please note these output files are the same format as the `default-labels.toml`
+file, so you can use one as a template if you have a repository set up with
+the labels you want organization-wide.
+
+```console
+# Make the directory if it doesn't already exist
+# The -p option specifies to create intermediate directories as required
+mkdir -p repo-labels-backup
+
+# Use the output of list-all-repos.py to fetch each repository's labels
+python list-all-repos.py | xargs -I {} labels fetch --owner $LABELS_ORGNAME --repo {} --filename repo-labels-backup/{}-labels.toml
+```
+
+### Apply default labels across the organization ###
+
+View and edit `default-labels.toml` to specify the default labels and their
+attributes such as name, color, and description. This is the same format as
+the output `.toml` files from the "back up" step, so if you have an example
+repository, you can copy straight from that repository's file.
+
+Please note that the `labels ... sync` operation is **destructive** and will
+reduce the set of labels in each repository to _only_ those specified.
+
+The label history of each issue and PR will show the removal of the previous
+labels in case you need to restore some.
+
+Once you've set up your `default-labels.toml` file, create a list of all the
+repositories in a file so you can edit the list to remove any repositories you
+do not want to alter labels for.
+
+```console
```

Comment:
```suggestion
```sh
```

Reply:
This is command-line usage so I believe `console` is more accurate.

Reply:
It's pointless to argue this miniscule point, but I will explain my (possibly insane) reasoning in case it makes any sense to you and should be added to our documented style.

I think it makes sense to use the formatting that best highlights the syntax in the code block.  I think `console` is most effective when showing the command _and_ its output, like this:
```console
$ the_first_command
the output
more output
$ the_second_command
yet more output
```
If I'm only showing the command (without the output) then I tend to prefer `sh` or `bash` if there is any shell-fu (pipes, file redirects, shell variables, etc.); otherwise it's a tossup between `console` and `sh` or `bash`.

Reply:
This is really great, thanks for writing it up! The current level of "documentation" consists of @mcdonnnj and your opinions from a text chat we had and this fleshes out your reasoning. I'll add it to the draft for now and we can issue a final ruling as a team once we've been able to split the hairs. 🥇

Reply:
Thanks @hillaryj!